java/src/jmri/jmrit/beantable/BeanTableDataModel.java

```diff
@@ -325,7 +365,12 @@ public void setValueAt(Object value, int row, int col) {
                 deleteBean(row, col);
                 break;
             default:
-                break;
+                NamedBeanPropertyDescriptor desc = getPropertyColumnDescriptor(col);
+                if (desc == null) {
+                    break;
+                }
+                NamedBean b = getBySystemName(sysNameList.get(row));
+                b.setProperty(desc.propertyKey, desc.renderProperty(value));
```

Comment:
This needs to be `b.setProperty(value);` to be correct, since passing the edited value through `renderProperty()` causes the property type to change to String, instead of being the type that was originally used.

Reply:
done